### PR TITLE
Ares Nuke Request

### PR DESCRIPTION
**Pull request by Ben10083**
# About the pull request

Implements ARES Emergency Procedure 'Request Nuke', which allows someone to petition High Command (Admins) for a nuke

# Explain why it's good for the game

This is already implemented in the framework for Project ARES, this just adds the feature. Should be fine as admins will need to authorize it. If people want a different rank for such nuke requests I can have it done. 

# Changelog
:cl:
add: You can now petition High Command via ARES for a nuclear device, bypassing techweb (If you can convince them, that is.)
admin: Implemented Admin side of ARES Nuke request to allow for granting of different nuke types, or denying the request.
/:cl:

### CODE REVIEW

**Comment by Ben10083** [comment]
it fucked up PAIN

**Comment by Ben10083** [comment]
Ignore the thousands of commits, it is now Ready™

File: code/modules/admin/topic/topic.dm [hunk 1]
```diff
@@ -1926,6 +1926,38 @@
 			log_game("[key_name_admin(usr)] has granted self-destruct, requested by [key_name_admin(ref_person)]")
 			message_admins("[key_name_admin(usr)] has granted self-destruct, requested by [key_name_admin(ref_person)]", 1)
 
+	if(href_list["nukeapprove"])
+		var/mob/ref_person = locate(href_list["nukeapprove"])
```
Comment (harryob) [inline code/modules/admin/topic/topic.dm:1930]: check the located ref is the correct type 
```
if(!istype(ref_person))
    return
```

File: code/modules/admin/topic/topic.dm [hunk 2]
```diff
@@ -1926,6 +1926,38 @@
...
+
+		//Can no longer request a nuke
+		GLOB.ares_link.interface.nuke_available = FALSE
+
+		marine_announcement("A nuclear device has been authorized by High Command and will be delivered to requisitions via ASRS.", "NUCLEAR ORDNANCE AUTHORIZED", 'sound/misc/notice2.ogg', logging = ARES_LOG_MAIN)
+		log_game("[key_name_admin(usr)] has authorized a [nuketype], requested by [key_name_admin(ref_person)]")
+		message_admins("[key_name_admin(usr)] has authorized a [nuketype], requested by [key_name_admin(ref_person)]")
+
+	if(href_list["nukedeny"])
+		var/mob/ref_person = locate(href_list["nukedeny"])
```
Comment (harryob) [inline code/modules/admin/topic/topic.dm:1956]: as above

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**